src/insert/spacefinder/rules.ts

```diff
@@ -0,0 +1,221 @@
+	window.guardian.config.page.hasShowcaseMainElement;
+
+const minDistanceBetweenRightRailAds = 500;
+const minDistanceBetweenInlineAds = isInHighValueSection ? 500 : 750;
+
+const candidateSelector = ':scope > p, [data-spacefinder-role="nested"] > p';
+
+const rightColumnOpponentSelector = '[data-spacefinder-role="immersive"]';
+const inlineOpponentSelector = [
+	'inline',
+	'supporting',
+	'showcase',
+	'thumbnail',
+	'richLink',
+]
+	.map((role) => `[data-spacefinder-role="${role}"]`)
+	.join(',');
+
+const headingSelector = ':scope > h2, [data-spacefinder-role="nested"] > h2';
+
+const desktopInline1: SpacefinderRules = {
+	bodySelector,
+	candidateSelector,
+	minAbove: isImmersive ? 700 : 300,
+	minBelow: 300,
+	opponentSelectorRules: {
+		// don't place ads right after a heading
+		[headingSelector]: {
+			minAboveSlot: isInHighValueSection ? 150 : 5,
+			minBelowSlot: isInHighValueSection ? 0 : 190,
+		},
+		[adSlotContainerSelector]: {
+			minAboveSlot: 500,
+			minBelowSlot: 500,
+		},
+		[inlineOpponentSelector]: {
+			minAboveSlot: 35,
+			minBelowSlot: 400,
+		},
+		[rightColumnOpponentSelector]: {
+			minAboveSlot: 35,
+			minBelowSlot: 400,
+		},
+	},
+};
+
+const desktopRightRailMinAbove = () => {
+	const base = 1000;
+	/**
+	 * In special cases, inline2 can overlap the "Most viewed" island, so
+	 * we need to make an adjustment to move the inline2 further down the page
+	 */
+	if (isPaidContent || !hasImages) {
+		return base + MOST_VIEWED_HEIGHT;
+	}
+
+	if (hasShowcaseMainElement) {
+		return base + 100;
+	}
+	return base;
+};
+
+const desktopRightRail: SpacefinderRules = {
+	bodySelector,
+	candidateSelector,
+	minAbove: desktopRightRailMinAbove(),
+	minBelow: 300,
+	opponentSelectorRules: {
+		[rightColumnOpponentSelector]: {
+			minAboveSlot: 0,
+			minBelowSlot: 600,
+		},
+	},
+	/**
+	 * Filter out any candidates that are too close to the last winner
+	 * see https://github.com/guardian/commercial/tree/main/docs/spacefinder#avoiding-other-winning-candidates
+	 * for more information
+	 **/
+	filter: (candidate, lastWinner) => {
+		if (!lastWinner) {
+			return true;
+		}
+		const largestSizeForSlot = adSizes.halfPage.height;
+		const distanceBetweenAds =
+			candidate.top - lastWinner.top - largestSizeForSlot;
+		return distanceBetweenAds >= minDistanceBetweenRightRailAds;
+	},
+};
+
+const mobileMinDistanceFromArticleTop = 200;
+
+const mobileCandidateSelector =
+	':scope > p, :scope > h2, :scope > [data-spacefinder-type$="NumberedTitleBlockElement"], [data-spacefinder-role="nested"] > p';
+
+const mobileHeadingSelector =
+	':scope > h2, [data-spacefinder-role="nested"] > h2, :scope > [data-spacefinder-type$="NumberedTitleBlockElement"]';
+
+const mobileOpponentSelectorRules = {
+	// don't place ads right after a heading
+	[mobileHeadingSelector]: {
+		minAboveSlot: 100,
+		minBelowSlot: 0,
+	},
+	[adSlotContainerSelector]: {
+		minAboveSlot: minDistanceBetweenInlineAds,
+		minBelowSlot: minDistanceBetweenInlineAds,
+	},
```

Comment:
I haven't seen an `adSlotContainerSelector` used for the old code for `mobileSubsequentInlineAds` also the value for these will be `500` or `750`. I haven't noticed any high numbers like that for mobile breakpoint. Could you please explain why we have it?